src/inmanta/deploy/tasks.py

```diff
@@ -0,0 +1,165 @@
+"""
+    Copyright 2024 Inmanta
+
+    Licensed under the Apache License, Version 2.0 (the "License");
+    you may not use this file except in compliance with the License.
+    You may obtain a copy of the License at
+
+        http://www.apache.org/licenses/LICENSE-2.0
+
+    Unless required by applicable law or agreed to in writing, software
+    distributed under the License is distributed on an "AS IS" BASIS,
+    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+    See the License for the specific language governing permissions and
+    limitations under the License.
+
+    Contact: code@inmanta.com
+"""
+
+import abc
+import datetime
+import logging
+import traceback
+import uuid
+from dataclasses import dataclass
+from typing import Optional, TypeAlias
+
+import inmanta.deploy.work as work
+from inmanta import const, data
+from inmanta.agent import executor
+from inmanta.deploy import scheduler, state
+
+
+class PoisonPill(work.Task):
+    """
+    Task to signal queue shutdown
+
+    It is used to make sure all workers wake up to observe that they have been closed.
+    It functions mostly as a no-op
+    """
+
+    async def execute(self, scheduler: "scheduler.ResourceScheduler", agent: str) -> None:
+        pass
+
+
+class WithHashMatchTask(work.Task):
```

Comment:
Where does this name come from? i.e. how does `RefreshFact` care about hash matching? From the implementation I see only the fact that they act on the latest version.

Reply:
It may not be the best name, but given the plan to eliminate it, I would not invest too much time in it

Reply:
> the plan to eliminate it

Do you mean given my proposal on Slack? I'm not sure it would go in its entirety but we'll see.

Reply:
I mean that